unit/util/irep_serialization.test.cpp

```diff
@@ -0,0 +1,188 @@
+/*******************************************************************
+ Module: irep serialization unit test
+
+ Author: Rafael Sá Menezes
+
+ Date: February 2020
+
+ Test Plan:
+   - Constructors
+   - Conversions
+ \*******************************************************************/
+
+#define BOOST_TEST_MODULE "irep Serialization"
+
+#include <util/irep_serialization.h>
+#include <util/irep2.h>
+#include <util/irep2_type.h>
+#include <boost/test/included/unit_test.hpp>
+#include <boost/test/framework.hpp>
+#include <util/irep2_expr.h>
+#include <util/migrate.h>
+
+namespace btf = boost::unit_test::framework;
+// THIS APPROACH IS NOT THREAD SAFE!!!!
```

Comment:
What are the consequences here?

Reply:
Our test framework is single thread. So it shouldn't have any problems. But since it is a helper function I preffered to add a warning.